### PR TITLE
Add appropriate arg suggestions in function call.

### LINKED ISSUE #1771: Intellisense:  Collect and tabular functions should only suggest tables
The first argument of collect shows a bunch of things. It should be limited to tables:

image

### PR DESCRIPTION
**Pull request by jas-valgotar**
Fixes #1771

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Utils/Extensions.cs [hunk 1]
```diff
@@ -91,5 +96,60 @@ private static void AddSeparator(StringBuilder sb, int depth)
...
+                    {
+                        errors.EnsureError(DocumentErrorSeverity.Severe, arg, TexlStrings.ErrTypeError_Arg_Expected_Found, name, dsNameType.GetKindString(), type.GetKindString());
+                    }
+
+                    isValid = false;
+                }
+            }
+
+            return isValid;
+        }
```
Comment (jas-valgotar) [inline src/libraries/Microsoft.PowerFx.Core/Utils/Extensions.cs:153]: Moved from Interpreter to Core

File: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseData/IntellisenseData.cs [hunk 2]
```diff
@@ -314,6 +313,7 @@ internal virtual void AddCustomSuggestionsForValuePossibilities()
         /// </summary>
         internal virtual void AddCustomSuggestionsForGlobals()
         {
+            AddSuggestionForCurrentFunction();
         }
```
Comment (jas-valgotar) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseData/IntellisenseData.cs:317]: This is overridden in PA, so the change doesn't impact PowerApps

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — DISMISSED** [review]
:shipit:

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**